src/AutoMapper/XpressionMapper/Extensions/MapperExtensions.cs

```diff
@@ -176,7 +176,18 @@ public static List<ParameterExpression> GetDestinationParameterExpressions(this
         public static Dictionary<Type, Type> AddTypeMapping<TSource, TDest>(this Dictionary<Type, Type> typeMappings)
             => typeMappings == null
                 ? throw new ArgumentException(Resource.typeMappingsDictionaryIsNull)
-                : typeMappings.AddTypeMapping(typeof(TSource), typeof(TDest));
+                : typeMappings.AddTypeMapping(typeof(TSource), typeof(TDest));
+
+        private static void AddUnderlyimgArrayElement(this Dictionary<Type, Type> typeMappings, Type sourceType, Type destType)

```

Comment:
Typo in the function name - should be AddUnderlyingArrayElement (same for the next one: AddUnderlyingGenericTypes)